python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -172,6 +178,73 @@ class OpSchedulingStatus:
     under_resource_limits: bool = False
 
 
+@dataclass
+class OpDisplayMetrics:
+    """Metrics of an operator. Used for display purposes."""
+
+    cpu: float = 0.0
+    gpu: float = 0.0
+    object_store_memory: float = 0.0
+    tasks: int = 0
+    actors: int = 0
+    queued: int = 0
+    task_backpressured: bool = False
+    task_backpressure_policies: List[str] = None
+    output_backpressured: bool = False
+    output_backpressure_policies: List[str] = None
+    extra_info: str = ""
+
+    def __post_init__(self):
+        if self.task_backpressure_policies is None:
+            self.task_backpressure_policies = []
+        if self.output_backpressure_policies is None:
+            self.output_backpressure_policies = []
+
+    def _format_policy_names(self, policies: List[str]) -> str:
+        """Format policy names by removing the 'BackpressurePolicy' suffix."""
+        return ",".join(p.replace("BackpressurePolicy", "") for p in policies)
+
+    def display_str(self) -> str:
+        """Format metrics object to a displayable string."""
+        metrics = []
+        # resource metrics
+        gpu_str = f" {self.gpu:.1f} GPU," if self.gpu else ""
+        mem_str = memory_string(self.object_store_memory)
+        metrics.append(f"{self.cpu:.1f} CPU,{gpu_str} {mem_str} object store")
+        # task
+        task_str = f"Tasks: {self.tasks}"
+        if self.task_backpressured or self.output_backpressured:
+            backpressured = []
+            if self.task_backpressured:
+                # Show which policies triggered task backpressure
+                if self.task_backpressure_policies:
+                    policy_names = self._format_policy_names(
+                        self.task_backpressure_policies
+                    )
```

Comment:
nit, formatting can be done when setting the attribute, instead of in every loop iteration.